src/extern-sync/SyncContextExtensions.cs

```diff
@@ -27,35 +31,280 @@ public static async Task<NpgsqlConnection> GetAndOpenDbConnectionAsync(this Bdms
+    {
+        foreach (var item in items)
+        {
+            item.ProcessRecursive(identifyable => identifyable.Id = 0, []);
+        }
+    }
+
+    /// <summary>
+    /// Recursively traverses the given <paramref name="item"/> and all its properties and
+    /// collections of type <typeparamref name="T"/> and executes <paramref name="action"/> on every object.
+    /// </summary>
+    /// <typeparam name="T">The type which should be traversed recursively.</typeparam>
+    /// <param name="item">The item to be processed recursively.</param>
+    /// <param name="action">The action to be executed on every <typeparamref name="T"/> object.</param>
+    /// <param name="visited">An empty <see cref="HashSet{T}"/> which is used to track the visited items
+    /// in order to avoid infinite loops.</param>
+    /// <returns>The updated <paramref name="item"/>.</returns>
+    internal static T ProcessRecursive<T>(this T item, Action<T> action, HashSet<T> visited)
+        where T : class
+    {
+        if (item == null || visited.Contains(item)) return item!;
+
+        visited.Add(item);
+
+        action(item);
+
+        foreach (var property in item.GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance))
+        {
+            if (!property.CanRead) continue;
+
+            var propertyValue = property.GetValue(item);
+            if (propertyValue == null) continue;
+
+            if (property.PropertyType.GetInterfaces().Contains(typeof(T)))
+            {
+                // If property itself implements T
+                ProcessRecursive((T)propertyValue, action, visited);
+            }
+            else if (typeof(IEnumerable).IsAssignableFrom(property.PropertyType))
+            {
+                foreach (var collectionItem in ((IEnumerable)propertyValue).OfType<T>())
+                {
+                    // If the property is an IEnumerable of T objects
+                    ProcessRecursive(collectionItem, action, visited);
+                }
+            }
+        }
+
+        return item;
+    }
+
+    /// <summary>
+    /// Sets the publication <paramref name="status"/> for the given <see cref="Borehole"/>.
+    /// Please note that all previous <see cref="Workflow"/> entries/histroy gets cleared!
+    /// This may not be the desired behavior for every use case but complies with the requirements
+    /// when syncing <see cref="Borehole"/>s from a context to another in this project.
+    /// </summary>
+    /// <param name="borehole">The <see cref="Borehole"/> to set the publication status on.</param>
+    /// <param name="status">The <see cref="Role"/>/Status to be set. <see cref="Role.View"/>
+    /// is not a valid publication status.</param>
+    /// <exception cref="NotSupportedException">If the <paramref name="status"/> is not supported.</exception>
+    internal static Borehole SetBoreholePublicationStatus(this Borehole borehole, Role status)
```

Comment:
REQ; Da der Parameter `status` aktuell nur einen Wert annehmen kann, also `Role.Publisher`, würde ich den _dynamischen_ Ansatz hier weglassen (→ ein bisschen YAGNI). Das würde die Implementation sehr vereinfachen und man könnte die Methode zu `SetBoreholeAsPublished` (oder so) umbenennen, was das Ganze nochmals verständlicher macht.

Nur wenn meine Annahme von oben korrekt ist.

Reply:
Da die Unterstützung der verschiedenen Status nur in den Tests verwendet wird habe ich den Code dorthin verschoben und für den Sync-Task eine spezifische Methode erstellt. Ich weiss das reduziert den Code nicht, gibt aber Klarheit wo was verwendet wird. Ich seh grad nicht wie ich den Aspekt sinnvoll rausnehmen kann ohne bei den Tests auf die unterschiedlichen Status bei den Bohrungen zu verzichten.

Ich denke diesen Aspekt (Lesen und Setzen der Borehole-Status under Berücksichtiung des Editmodes) müssen wir im Kontext der Konsiliderungsphase angehen. Dieser Bereich soll ja sowieso überarbeitet/angepasst werden.

Können wir sonst aber gerne nochmals diskutieren.